src/smirnoffio.py

```diff
@@ -383,12 +434,21 @@ def prepare(self, pbc=False, mmopts={}, **kwargs):
             else:
                 box_omm = None
             # Finally append it to list.
-            self.xyz_omms.append((position, box_omm))
+            self.xyz_omms.append((xyz_omm, box_omm))
+
+        # used in create_simulation()
+        openmm_topology = SMIRNOFF._openff_to_openmm_topology(openff_topology)
+        openmm_positions = (
+            self.pdb.positions.value_in_unit(angstrom) +
+            # Add placeholder positions for an v-sites.
+            [Vec3(0.0, 0.0, 0.0)] * openff_topology.n_topology_virtual_sites
+        ) * angstrom
```

Comment:
Also it looks like the positions of virtual sites are always appended to the end of the entire structure. Does this ever lead to issues?

Reply:
I think the OpenFF toolkit made the decision to add virtual sites at the end of a topology so that the indices of atoms were preserved. @j-wags can probably comment on that a bit more though. As far I can tell this doesn't cause issues, although it is a little more awkward to work with.